src/github.com/t3chguy/riot-static/riot-static.go

```diff
@@ -0,0 +1,234 @@
+	var page int
+	var err error
+	if page, err = strconv.Atoi(c.DefaultQuery("page", "1")); err != nil {
+		page = 1
+	}
+
+	pageSize := 20
+
+	data.RLock()
+	skip, end := paginate(page, pageSize, data.NumRooms)
+	c.HTML(http.StatusOK, "rooms.html", gin.H{
+		"Rooms":    data.Ordered[skip:end],
+		"NumRooms": data.NumRooms,
+		"Page":     page,
+	})
+	data.RUnlock()
+}
+
+func GetPublicRoom(c *gin.Context) { // ~/room/:roomId/chat
+	roomId := c.Param("roomId")
+
+	data.RLock()
+	c.HTML(http.StatusOK, "room.html", gin.H{
+		"Room": data.Rooms[roomId],
+	})
+	data.RUnlock()
+}
+
+func GetPublicRoomServers(c *gin.Context) { // ~/room/:roomId/servers
+	roomId := c.Param("roomId")
+
+	data.RLock()
+	c.HTML(http.StatusOK, "room_servers.html", gin.H{
+		"Room": data.Rooms[roomId],
+	})
+	data.RUnlock()
+}
+
+func GetPublicRoomMembers(c *gin.Context) { // ~/room/:roomId/members
+	roomId := c.Param("roomId")
+
+	//var page int
+	//var err error
+	//if page, err = strconv.Atoi(c.DefaultQuery("page", "1")); err != nil {
+	//	page = 1
+	//}
+
+	//pageSize := 20
+
+	data.RLock()
+	length := len(data.Rooms[roomId].MemberInfo)
+	//skip, end := paginate(page, pageSize, length)
+	c.HTML(http.StatusOK, "room_members.html", gin.H{
+		"Room": data.Rooms[roomId],
+		//"MemberInfo": data.Rooms[roomId].Members[skip:end],
+		"NumMembers": length,
+	})
+	data.RUnlock()
+}
+
+func GetPublicRoomPowerLevels(c *gin.Context) { // ~/room/:roomId/power_levels
+	roomId := c.Param("roomId")
+
+	data.RLock()
+	c.HTML(http.StatusOK, "power_levels.html", gin.H{
+		"PowerLevels": data.Rooms[roomId].PowerLevels,
+	})
+	data.RUnlock()
+}
+
+func GetPublicRoomMember(c *gin.Context) { // ~/room/:roomId/members/:mxid
+	roomId := c.Param("roomId")
+	mxid := c.Param("mxid")
+
+	data.RLock()
+	if memberInfo := data.Rooms[roomId].MemberInfo[mxid]; memberInfo != nil {
+		c.HTML(http.StatusOK, "member_info.html", gin.H{
+			"RoomID":     roomId,
+			"MXID":       mxid,
+			"MemberInfo": memberInfo,
+		})
+	} else {
+		c.AbortWithStatus(http.StatusNotFound)
+	}
+	data.RUnlock()
+}
+
+var data = struct {
+	resync.Once
+	sync.RWMutex
+	NumRooms int
+	Ordered  []*Room
+	Rooms    map[string]*Room
+}{}
+
+func LoadPublicRooms() {
+	// @TODO: fix this.
+	data.Lock()
+	fmt.Println("Loading public publicRooms")
+	resp, err := cli.PublicRooms(0, "", "")
+
+	if err == nil {
+		b := []*Room{}
+		c := map[string]*Room{}
+
+		// filter on actually WorldReadable publicRooms
+		for _, x := range resp.Chunk {
+			if x.WorldReadable {
+				var room *Room
+				if data.Rooms[x.RoomId] != nil {
+					room = data.Rooms[x.RoomId]
+				} else {
+					room = NewRoom(x)
+				}
+				b = append(b, room)
+				c[x.RoomId] = room
+			}
+		}
+
+		//data.Lock()
+		data.Rooms = c
+		data.NumRooms = len(b)
+		// copy order so we don't encounter slice hell
+		data.Ordered = make([]*Room, data.NumRooms)
+		copy(data.Ordered, b)
+
+		data.Unlock()
+	}
+
+	if err != nil {
+		panic(err)
+	}
+}
+
+func main() {
+	setupCli()
```

Comment:
Naming: CLI == Command Line Interface, but what I suspect you meant was `setupClient()`. I would personally move this function to be just above the `main()` function because it:
 - Is only called in one place, right here.
 - Cannot be used without risking an entire process exit, which only makes sense in the context of starting up.